diff --git a/Makefile b/Makefile
index 1c0eddc..9daf2dc 100644
--- a/Makefile
+++ b/Makefile
@@ -58,7 +58,7 @@ endif
 
 
 DEFINES     += $(addprefix -I$(SOURCE_PATH),. zstd/lib zstd/lib/common brotli/include xpack/common libcsc)
-DEFINES     += -DHAVE_CONFIG_H
+DEFINES     += -DHAVE_CONFIG_H -I/usr/include/c++/7/ -I/usr/aarch64-linux-gnu/include/c++/7/aarch64-linux-gnu/ -I/usr/aarch64-linux-gnu/include
 CODE_FLAGS  += -Wno-unknown-pragmas -Wno-sign-compare -Wno-conversion
 OPT_FLAGS   ?= -fomit-frame-pointer -fstrict-aliasing -ffast-math
 
@@ -67,13 +67,13 @@ ifeq ($(BUILD_TYPE),debug)
 	OPT_FLAGS_O2 = $(OPT_FLAGS) -g
 	OPT_FLAGS_O3 = $(OPT_FLAGS) -g
 else
-	OPT_FLAGS_O2 = $(OPT_FLAGS) -O2 -DNDEBUG
-	OPT_FLAGS_O3 = $(OPT_FLAGS) -O3 -DNDEBUG
+	OPT_FLAGS_O2 = $(OPT_FLAGS) -O2 -DNDEBUG --target=aarch64-linux-gnu -D__STRICT_ANSI__
+	OPT_FLAGS_O3 = $(OPT_FLAGS) -O3 -DNDEBUG --target=aarch64-linux-gnu -D__STRICT_ANSI__
 endif
 
 CFLAGS = $(MOREFLAGS) $(CODE_FLAGS) $(OPT_FLAGS_O3) $(DEFINES)
 CFLAGS_O2 = $(MOREFLAGS) $(CODE_FLAGS) $(OPT_FLAGS_O2) $(DEFINES)
-LDFLAGS += $(MOREFLAGS)
+LDFLAGS += $(MOREFLAGS) --target=aarch64-linux-gnu 
 
 
 LZO_FILES = lzo/lzo1.o lzo/lzo1a.o lzo/lzo1a_99.o lzo/lzo1b_1.o lzo/lzo1b_2.o lzo/lzo1b_3.o lzo/lzo1b_4.o lzo/lzo1b_5.o
